### PR TITLE
Fix #28

### LINKED ISSUE #28: It roll when I use `this.setState()` (Class Components)
As the title, the `<AnimatedNumbers />` roll again when I use the `this.setState()` function even though they are not in the same components. (the `animateToNumber` value dons't change).

For Example, 

```jsx
import AnimatedNumber from "animated-number-react";
import React, {useState} from "react";
import AnimatedNumbers from "react-animated-numbers";
import {Button} from "@mui/material";

export default class ShowArkBoxIDNumber extends React.Component{
    constructor(props) {
        super(props);
        this.state = {
            a: 123456,
            b: 999
        }
    }

    render() {
        return (
            <div>
                <AnimatedNumbers
                    animateToNumber={this.state.a}
                    fontStyle={{ fontSize: 32 }}
                    configs={(number, index) => {
                        return { mass: 1, tension: 230 * (index + 1), friction: 140 };
                    }}
                ></AnimatedNumbers>
                <Button id={"Button 1"} onClick={() => this.setState({a: Math.round(Math.random() * 1000)})}>Button 1</Button>
                <Button id={"Button 2"} onClick={() => this.setState({b: 123})}>Button 2</Button>
            </div>
        );
    }
}
```
In this components, 
If I click Button 1: it roll and change number.
If I click Button 2: it roll too but donut change number.
Is there any solution?

### PR DESCRIPTION
**Pull request by WhiteNightAWA**

### CODE REVIEW

File: index.jsx [hunk 1]
```diff
@@ -117,6 +117,8 @@ const AnimatedNumber = ({
   );
 };
 
-const Enhanced = React.memo(AnimatedNumber)
+const Enhanced = React.memo(AnimatedNumber, (prevProps, nextProps) => {
+    return prevProps.animateToNumber === nextProps.animateToNumber
```
Comment (heyman333) [inline index.jsx:121]: Since **fontstyle** can also be a prop, I think it would be better to add a dependency for it as well.

**Review by heyman333 — COMMENTED** [review]
Thank you for you pr, I hope you fix one thing

**Comment by WhiteNightAWA** [comment]
@heyman333 Just done :)

**Review by heyman333 — APPROVED** [review]
Thanks!

**Pull request merged**